File: string/firstUniqueCharacterInAString.js
// https://leetcode.com/problems/first-unique-character-in-a-string/
/**
 * @param {string} s
 * @return {number}
 */
const firstUniqChar = (s) => {
  const map = new Map();
  for (let index = 0; index < s.length; index++) {
    if (map.has(s[index])) {
      map.set(s[index], -1);
    } else {
      map.set(s[index], index);
    }
  }
  let firstUniqIndex = -1;
  for (const [key, index] of map) {
    if (index != -1) {
      firstUniqIndex = index;
      break;
    }
  }
  return firstUniqIndex;
};

File: string/firstUniqueCharacterInAString.py
# https://leetcode.com/problems/first-unique-character-in-a-string/
class Solution:
    def firstUniqChar(self, s: str) -> int:
        mapping = {}
        for index, char in enumerate(s):
            if char in mapping.keys():
                mapping[char] = -1
            else:
                mapping[char] = index

        for char in mapping:
            if mapping[char] > -1:
                return mapping[char]

        return -1
